### PR TITLE
Update no-deps branch

### PR DESCRIPTION
**Pull request by thealmarty**

### CODE REVIEW

**Comment by thealmarty** [comment]
- the cargo test PR isn't merged into here because I haven't tested for it to work atm.

**Comment by marcinbugaj** [comment]
Related to https://github.com/lita-xyz/valida-toolchain/issues/564

File: README.md [hunk 1]
```diff
@@ -1,7 +1,42 @@
...
+## The `entrypoint!` macro
+
+The `entrypoint!` macro:
+
+- Sets up a deterministic random number generator: It ensures that when `rand` functions are called, they are fixed to a specified seed and thus are deterministic. This is required for the program to be provable.
+- Creates a new entry point that wraps the user's main function: This is required because we need to make Rust call this `main` function, the standard Rust `main` function does not work on Valida.
+
+### For projects that require `rand`
+
+Use the `main` branch if you need randomness. Add the following to your `src/main.rs`:
```
[the remark this answers is not in the document]
Comment (thealmarty) [reply README.md:14]: huh now sure how those files got in there. Fixed in the latest commit. Thanks!

**Review by lewis1revill — APPROVED** [review]
Approved, we'll look into using cargo features at a later date

**Pull request merged**